CS483/CS483/Kartaclysm/Components/Abilities/ComponentWheelieAbility.cpp

```diff
@@ -0,0 +1,113 @@
+//----------------------------------------------------------------------------
+// ComponentWheelieAbility.h
+// Author: Bradley Cooper
+//
+// Showoff's ability to reduce steering but gain speed.
+//----------------------------------------------------------------------------
+
+#include "ComponentWheelieAbility.h"
+
+namespace Kartaclysm
+{
+	ComponentWheelieAbility::ComponentWheelieAbility(
+		HeatStroke::GameObject* p_pGameObject,
+		int p_iStrength,
+		float p_fStatusDuration)		
+		:
+		ComponentAbility(p_pGameObject),
+		m_iStrength(p_iStrength),
+		m_fStatusDuration(p_fStatusDuration)
+	{
+	}
+
+	ComponentWheelieAbility::~ComponentWheelieAbility()
+	{
+		HeatStroke::EventManager::Instance()->RemoveListener(GetGameObject()->GetGUID(), m_pAbilityDelegate);
+		delete m_pAbilityDelegate;
+		m_pAbilityDelegate = nullptr;
+	}
+
+	HeatStroke::Component* ComponentWheelieAbility::CreateComponent(
+		HeatStroke::GameObject* p_pGameObject,
+		tinyxml2::XMLNode* p_pBaseNode,
+		tinyxml2::XMLNode* p_pOverrideNode)
+	{
+		assert(p_pGameObject != nullptr);
+
+		// Defaults
+		int iStrength = 0;
+		float fStatusDuration = 0.0f;
+
+		// All parameters are optional.
+		if (p_pBaseNode != nullptr)
+		{
+			ParseNode(p_pBaseNode, iStrength, fStatusDuration);
+		}
+		if (p_pOverrideNode != nullptr)
+		{
+			ParseNode(p_pOverrideNode, iStrength, fStatusDuration);
+		}
+
+		// Check that we got everything we needed.
+		assert(iStrength != 0);
+		assert(iStrength != 0.0f);
```

Comment:
Two things:
1. Maybe I'm missing something, but on line 41 you say all parameters are optional, then here you assert that the values aren't still the defaults, which seems to indicate that they aren't optional.
2. On line 53, is this supposed to be `assert(fStatusDuration != 0.0f);`?